sdk/containerregistry/container-registry/review/container-registry.api.md

```diff
@@ -57,6 +57,20 @@ export interface ArtifactTagProperties {
     readonly repositoryName: string;
 }
 
+// @public
+export class ContainerRegistryBlobClient {
+    constructor(endpoint: string, repositoryName: string, credential: TokenCredential, options?: ContainerRegistryClientOptions);
+    deleteBlob(digest: string, options?: DeleteBlobOptions): Promise<void>;
+    deleteManifest(digest: string, options?: DeleteManifestOptions): Promise<void>;
+    downloadBlob(digest: string, options?: DownloadBlobOptions): Promise<DownloadBlobResult>;
+    downloadManifest(tagOrDigest: string, options?: DownloadManifestOptions): Promise<DownloadManifestResult>;
+    readonly endpoint: string;
+    readonly repositoryName: string;
+    uploadBlob(blob: NodeJS.ReadableStream): Promise<UploadBlobResult>;
```

Comment:
@bterlson should we get in the habit of supporting accepting a function that returns a `ReadableStream` in addition to places where we accept a `ReadableStream` directly?

Reply:
FWIW, this would be helpful in this exact scenario, actually -- at the moment we will most likely have to consume the stream twice: once to calculate the digest we send to the server, and once to actually upload the blob. The other languages are achieving this by just copying the stream into memory in its entirety and reading it twice, but we could avoid that in the case that a `ReadableStream` factory function is passed in, since we can just call the function twice and get two streams. But I guess that would only work if customers understood that their function could get called multiple times.

Reply:
Yeah, it's a good point, I'm not familiar enough with this kind of scenario to have good intuition. @jeremymeng this shows up in storage-blob right? What are your thoughts?

Reply:
Yes, I think taking a function returning `ReadableStream` is better for this scenario for the reason @timovv mentioned.  Currently storage blob's `upload` supports core-http request body type which include the factory function, while `uploadStream` only takes a `Readable`. I remember there were some discussions around using a factory function doesn't prevent customers from passing the same stream instance multiple times but might still be helpful.